planetscale/imports.go

```diff
@@ -0,0 +1,306 @@
+	Errors             string           `json:"import_check_errors"`
+	StartedAt          *time.Time       `json:"started_at"`
+	FinishedAt         *time.Time       `json:"finished_at"`
+	DeletedAt          *time.Time       `json:"deleted_at"`
+	ExternalDataSource DataImportSource `json:"data_source"`
+}
+
+func (di *DataImport) ParseState() {
+	if val, ok := stateToImportStateMap[di.State]; ok {
+		di.ImportState = val
+		return
+	}
+
+	panic("unknown data import state " + di.State)
+}
+
+type TestDataImportSourceRequest struct {
+	Organization string           `json:"organization"`
+	Database     string           `json:"database_name"`
+	Source       DataImportSource `json:"connection"`
+}
+
+// DataSourceIncompatibilityError represents an error that occurs when the
+// source schema in an external database server is incompatible with PlanetScale.
+type DataSourceIncompatibilityError struct {
+	LintError        string `json:"lint_error"`
+	Keyspace         string `json:"keyspace_name"`
+	Table            string `json:"table_name"`
+	SubjectType      string `json:"subject_type"`
+	ErrorDescription string `json:"error_description"`
+	DocsUrl          string `json:"docs_url"`
+}
+
+type TestDataImportSourceResponse struct {
+	CanConnect   bool                              `json:"can_connect"`
+	ConnectError string                            `json:"error"`
+	Errors       []*DataSourceIncompatibilityError `json:"lint_errors"`
+}
+
+type StartDataImportRequest struct {
+	Database     string           `json:"database_name"`
+	Organization string           `json:"organization"`
+	Source       DataImportSource `json:"connection"`
+}
+
+type MakePlanetScalePrimaryRequest struct {
+	Organization string
+	Database     string
+}
+
+type MakePlanetScaleReplicaRequest struct {
+	Organization string
+	Database     string
+}
+
+type DetachExternalDatabaseRequest struct {
+	Organization string
+	Database     string
+}
+
+type GetImportStatusRequest struct {
+	Organization string
+	Database     string
+}
+
+type CancelDataImportRequest struct {
+	Organization string
+	Database     string
+}
+
+// DataImportsService is an interface for communicating with the PlanetScale
+// Data Imports API endpoint.
+type DataImportsService interface {
+	TestDataImportSource(ctx context.Context, request *TestDataImportSourceRequest) (*TestDataImportSourceResponse, error)
+	StartDataImport(ctx context.Context, request *StartDataImportRequest) (*DataImport, error)
+	CancelDataImport(ctx context.Context, request *CancelDataImportRequest) error
+	GetDataImportStatus(ctx context.Context, request *GetImportStatusRequest) (*DataImport, error)
+	MakePlanetScalePrimary(ctx context.Context, request *MakePlanetScalePrimaryRequest) (*DataImport, error)
+	MakePlanetScaleReplica(ctx context.Context, request *MakePlanetScaleReplicaRequest) (*DataImport, error)
+	DetachExternalDatabase(ctx context.Context, request *DetachExternalDatabaseRequest) (*DataImport, error)
+}
```

Comment:
Could we add doc comments, at least for some of the important ones, such as Start and Make?